dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/datastore/hibernate/HibernateDatastoreStore.java

```diff
@@ -172,15 +174,138 @@ public DatastoreEntry getEntry(String namespace, String key) {
   }
 
   @Override
-  public void deleteNamespace(String namespace) {
-    String hql = "delete from DatastoreEntry v where v.namespace = :namespace";
-    getSession().createQuery(hql).setParameter("namespace", namespace).executeUpdate();
+  public void deleteNamespace(String ns) {
+    // language=SQL
+    String sql = "delete from keyjsonvalue ds where ds.namespace = :ns";
+    getSession().createNativeQuery(sql).setParameter("ns", ns).executeUpdate();
   }
 
   @Override
-  public int countKeysInNamespace(String namespace) {
-    String hql = "select count(*) from DatastoreEntry v where v.namespace = :namespace";
-    Query<Long> count = getTypedQuery(hql);
-    return count.setParameter("namespace", namespace).getSingleResult().intValue();
+  public int countKeysInNamespace(String ns) {
+    // language=SQL
+    String sql = "select count(*) from keyjsonvalue v where v.namespace = :ns";
+    Object count = getSession().createNativeQuery(sql).setParameter("ns", ns).uniqueResult();
+    if (count == null) return 0;
+    if (count instanceof Number n) return n.intValue();
+    throw new IllegalStateException("Count did not return a number but: " + count);
+  }
+
+  @Override
+  public boolean updateEntry(
```

Comment:
this method would read more easily/nicely if each flow was its own method, making it much easier to comprehend at the top level what this method does, without having to look through the implementation details for each flow.

Reply:
I'll make a cleanup PR

Reply:
https://github.com/dhis2/dhis2-core/pull/15899 :)